pay-api/src/pay_api/services/report_service.py

```diff
@@ -0,0 +1,58 @@
+# Copyright © 2024 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Service to manage report generation."""
+from dataclasses import dataclass
+
+from flask import current_app
+from pay_api.utils.user_context import user_context
+from .oauth_service import OAuthService
+from ..utils.enums import AuthHeaderType, ContentType
+
+
+@dataclass
+class ReportRequest:
+    """Used for reportAPI request payload."""
+
+    content_type: str
+    report_name: str
+    template_name: str
+    template_vars: dict
+    populate_page_number: bool
+
+
+class ReportService:
+    """Service to manage report generation."""
+
+    @staticmethod
+    def get_request_payload(request: ReportRequest) -> dict:
+        """Return report request payload dictionary."""
+        return {
```

Comment:
just return `humps.camelize(asdict(request))`

similar to 
![image](https://github.com/bcgov/sbc-pay/assets/3484109/fe68c48a-9dde-4f64-a769-50a13291670c)

Reply:
There is an issue with his as the outer report payload expects camel case, but a lot of the actual template_vars are snake case and used as such in existing templates due to how the values are retrieved.

Reply:
I still don't see the issue, you're expecting camelCase in this scenario

Reply:
I believe it converts the whole payload including statement attributes to camel as well, but existing templates use statement.from_date

```
"templateVars": {
  "statement": {
              "created_on": "2024-05-27",
              "frequency": "MONTHLY",
              "from_date": "2024-04-01",
              "to_date": "2024-04-30",
              ...
  }
```

Reply:
I see what you mean now and that's a huge pain to fix everywhere along the way